src/model/OutputTableSummaryReports.hpp

```diff
@@ -0,0 +1,114 @@
+*  disclaimer in the documentation and/or other materials provided with the distribution.
+*
+*  (3) Neither the name of the copyright holder nor the names of any contributors may be used to endorse or promote products
+*  derived from this software without specific prior written permission from the respective party.
+*
+*  (4) Other than as required in clauses (1) and (2), distributions in any form of modifications or other derivative works
+*  may not use the "OpenStudio" trademark, "OS", "os", or any other confusingly similar designation without specific prior
+*  written permission from Alliance for Sustainable Energy, LLC.
+*
+*  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDER(S) AND ANY CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+*  INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+*  DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER(S), ANY CONTRIBUTORS, THE UNITED STATES GOVERNMENT, OR THE UNITED
+*  STATES DEPARTMENT OF ENERGY, NOR ANY OF THEIR EMPLOYEES, BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+*  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF
+*  USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+*  STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+*  ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+***********************************************************************************************************************/
+
+#ifndef MODEL_OUTPUTTABLESUMMARYREPORTS_HPP
+#define MODEL_OUTPUTTABLESUMMARYREPORTS_HPP
+
+#include <vector>
+#include "ModelAPI.hpp"
+#include "ModelObject.hpp"
+
+namespace openstudio {
+namespace model {
+
+namespace detail {
+
+  class OutputTableSummaryReports_Impl;
+
+} // detail
+
+/** OutputTableSummaryReports is a ModelObject that wraps the OpenStudio IDD object 'OS:Output:Table:SummaryReports'. */
+class MODEL_API OutputTableSummaryReports : public ModelObject {
+ public:
+  /** @name Constructors and Destructors */
+  //@{
+
+  virtual ~OutputTableSummaryReports() {}
+
+  //@}
+
+  static IddObjectType iddObjectType();  
+
+  /** @name Getters */
+  //@{
+
+  void removeSummaryReport(int groupIndex);
+
+  void removeAllSummaryReports();
+
+  std::vector<std::string> summaryReports() const;
+
+  unsigned int numberofSummaryReports() const;
+
+  /** If a summaryReport group is already present (cf `summaryReportIndex()`), it will Warn */
+  boost::optional<unsigned> summaryReportIndex(std::string summaryReport) const;
+
+  boost::optional<std::string> getSummaryReport(unsigned groupIndex) const;
+
+  //@}
+  /** @name Setters */
+  //@{
+
+  bool addSummaryReport(std::string summaryReport);
+
+  bool addSummaryReports(std::vector<std::string> summaryReports);
```

Comment:
This API is fine (and has a bunch of priors). But I realize now that our model API is inconsistent with non-extensible fields.

@kbenne @joseph-robertson I think it would be worthwhile at one point to write guidelines on how to handle extensible fields. Our API is inconsistent from one object to another. Non-extensible fields are handled by GenerateClass.rb which ensures an appropriate level of standardization across classes, extensible fields aren't.

I've done something different recently for `OutputDiagnostics` for example, where I didn't expose groupIndex stuff, and preferred a `setXXXs(std::vector<std::string>&)` which would `clearExtensibleGroups` before adding new ones. I also called the "eraser" `clearXXX()` instead of `removeAllXXXs`

https://github.com/NREL/OpenStudio/blob/92c9f11baf755c907e5b4c07df367f4dc9dcbbef/src/model/OutputDiagnostics.hpp#L69-L80